config.go

```diff
@@ -44,10 +44,26 @@ func loadConfig() (*config, error) {
 		return nil, fmt.Errorf("unsupported overlay network size; net mask must be multiple of 8, got %d", bits)
 	}
 
-	// FIXME: this is a workaround for memberlist refusing to listen on public IPs if BindAddr==0.0.0.0
 	if config.BindAddr != "" && config.BindIface != "" {
 		return nil, fmt.Errorf("setting both bind address and bind interface is not supported")
+
+	} else if config.BindIface != "" {
+		// Compute the actual bind address based on the provided interface
+		iface, err := net.InterfaceByName(config.BindIface)
+		if err != nil {
+			return nil, fmt.Errorf("cannot find interface %s", config.BindIface)
```

Comment:
this is an improvement on the original version, but I think we should still wrap the original error and not assume what it's about:
```suggestion
			return nil, fmt.Errorf("could not get interface by name %s: %w", config.BindIface, err)
```